### PR TITLE
Improve installation documentation when using a tarball

### PR DESCRIPTION
**Pull request by Rombur**
The current instructions to install DTK do not work if we are downloading a tarball. 

I will cherry-pick this commit.

### CODE REVIEW

File: docs/source/install.rst [hunk 1]
```diff
@@ -65,6 +65,15 @@ Another way to achieve this is to pass the ``--recursive`` option to the ``git
...
 DataTransferKit repository.
 
+You can also download the DTKData
+`repository <https://github.com/ORNL-CEES/DTKData>`_ yourself and then, link
+it into DTK directory:
+
+.. code::
+
+  $ cd $DTK_DIR
+  $ ln -s $DTKDATA_DIR DTKData
```
Comment (dalg24) [inline docs/source/install.rst]: Would need to define env variable `DTKDATA_DIR` for that to work.
I realize now that there is the same problem with `DTK_DIR` but the goal would be that people can almost copy/paste from these instructions.

Comment (Rombur) [reply docs/source/install.rst]: Sure but I assumed that people were smart enough to replace the `$DTKDATA_DIR` with the right path.

Comment (dalg24) [reply docs/source/install.rst]: I usually copy and paste without thinking and add `sudo` in front of it if it didn't work.

Comment (Rombur) [reply docs/source/install.rst]: I'll make sure to add `rm -rf /` at the end of the instruction then.

Comment (dalg24) [reply docs/source/install.rst]:
```
root@00f6503f974c:/# rm -rf /
rm: it is dangerous to operate recursively on '/'
rm: use --no-preserve-root to override this failsafe
```

**Pull request merged**